File: src/main/java/uk/gov/justice/digital/hmpps/oauth2server/model/UserCaseloadRoleIdentity.java
package uk.gov.justice.digital.hmpps.oauth2server.model;

import lombok.*;

import javax.persistence.Column;
import javax.persistence.Embeddable;
import java.io.Serializable;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder(toBuilder = true)
@EqualsAndHashCode(of = {"username", "roleId", "caseload"})
@Embeddable
public class UserCaseloadRoleIdentity implements Serializable {

    @Column(name = "username")
    private String username;

    @Column(name = "role_id")
    private String roleId;

    @Column(name = "caseload_id")
    private String caseload;

}

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/nomis/model/UserCaseloadRoleIdentity.kt
package uk.gov.justice.digital.hmpps.oauth2server.nomis.model

import java.io.Serializable
import javax.persistence.Column
import javax.persistence.Embeddable

@Embeddable
data class UserCaseloadRoleIdentity(
  @Column(name = "role_id")
  val roleId: Long,

  @Column
  val username: String,

  @Column(name = "caseload_id")
  val caseload: String,
) : Serializable
